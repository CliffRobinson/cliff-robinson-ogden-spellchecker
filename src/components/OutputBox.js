export function OutputBox({wordArray}) {

    return (<div>
        {wordArray.map(word => {
            const styleObj = word.spelling ? {color:"black"} : {color:"red"}
            return <span style={styleObj}>{word.word} </span>
        })}
    </div>)
}