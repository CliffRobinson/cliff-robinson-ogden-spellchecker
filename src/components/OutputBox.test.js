import {shallow, mount} from 'enzyme'

import {OutputBox} from './OutputBox'

const quote = [
    { word: 'A', spelling: true },
    { word: 'wizard', spelling: undefined },
    { word: 'is', spelling: undefined },
    { word: 'never', spelling: undefined },
    { word: 'late,', spelling: true },
    { word: 'nor', spelling: undefined },
    { word: 'is', spelling: undefined },
    { word: 'he', spelling: true },
    { word: 'early,', spelling: true }
  ]

test('userinputbox render', () => {
  const tree = mount(<OutputBox wordArray={quote}/>)
  expect(tree).toMatchSnapshot()
})