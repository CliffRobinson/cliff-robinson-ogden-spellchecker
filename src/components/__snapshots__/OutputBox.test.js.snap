// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userinputbox render 1`] = `
<OutputBox
  wordArray={
    Array [
      Object {
        "spelling": true,
        "word": "A",
      },
      Object {
        "spelling": undefined,
        "word": "wizard",
      },
      Object {
        "spelling": undefined,
        "word": "is",
      },
      Object {
        "spelling": undefined,
        "word": "never",
      },
      Object {
        "spelling": true,
        "word": "late,",
      },
      Object {
        "spelling": undefined,
        "word": "nor",
      },
      Object {
        "spelling": undefined,
        "word": "is",
      },
      Object {
        "spelling": true,
        "word": "he",
      },
      Object {
        "spelling": true,
        "word": "early,",
      },
    ]
  }
>
  <div>
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      A
       
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      wizard
       
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      is
       
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      never
       
    </span>
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      late,
       
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      nor
       
    </span>
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      is
       
    </span>
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      he
       
    </span>
    <span
      style={
        Object {
          "color": "black",
        }
      }
    >
      early,
       
    </span>
  </div>
</OutputBox>
`;
